test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -110,252 +94,292 @@ public void addUnit() {
+    assertThat(unitCollection.isEmpty(), is(equalTo(false)));
   }
 
   @Test
   public void getUnits() {
     final Collection<Unit> unitsOfOtherPlayerOfUnitTypeOne = getUnitsOfOtherPlayerOfUnitTypeOne();
-    uc.addAllUnits(unitsOfOtherPlayerOfUnitTypeOne);
-    assertThat(uc.getUnits().containsAll(unitsOfOtherPlayerOfUnitTypeOne), is(equalTo(true)));
+    unitCollection.addAllUnits(unitsOfOtherPlayerOfUnitTypeOne);
+    assertThat(unitCollection.getUnits().containsAll(unitsOfOtherPlayerOfUnitTypeOne), is(equalTo(true)));
   }
 
   @Test
   public void getPlayersWithUnits() {
-    assertThat(uc.getPlayersWithUnits().size(), is(equalTo(0)));
-    addAllDefaultPlayerUnits();
-    assertThat(uc.getPlayersWithUnits().size(), is(equalTo(1)));
-    addAllOtherPlayerUnits();
-    assertThat(uc.getPlayersWithUnits().size(), is(equalTo(2)));
+    assertThat(unitCollection.getPlayersWithUnits().size(), is(equalTo(0)));
+    addAllDefaultPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.getPlayersWithUnits().size(), is(equalTo(1)));
+    addAllOtherPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.getPlayersWithUnits().size(), is(equalTo(2)));
   }
 
   @Test
   public void getPlayerUnitCounts() {
-    addAllPlayerUnits();
-    final IntegerMap<PlayerID> playerUnitCounts = uc.getPlayerUnitCounts();
+    addAllPlayerUnitsToUnitCollection();
+    final IntegerMap<PlayerID> playerUnitCounts = unitCollection.getPlayerUnitCounts();
     assertThat(playerUnitCounts.getInt(defaultPlayerID), is(equalTo(unitCountDefaultPlayer)));
     assertThat(playerUnitCounts.getInt(otherPlayerID), is(equalTo(unitCountOtherPlayer)));
   }
 
   @Test
   public void hasUnitsFromMultiplePlayers() {
-    addAllDefaultPlayerUnits();
-    assertThat(uc.hasUnitsFromMultiplePlayers(), is(equalTo(false)));
-    addAllOtherPlayerUnits();
-    assertThat(uc.hasUnitsFromMultiplePlayers(), is(equalTo(true)));
+    addAllDefaultPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.hasUnitsFromMultiplePlayers(), is(equalTo(false)));
+    addAllOtherPlayerUnitsToUnitCollection();
+    assertThat(unitCollection.hasUnitsFromMultiplePlayers(), is(equalTo(true)));
   }
 
   @Test
   public void getHolder() {
-    assertThat(uc.getHolder(), is(equalTo(defaultPlayerID)));
+    assertThat(unitCollection.getHolder(), is(equalTo(defaultPlayerID)));
   }
 
   @Test
   public void allMatch() {
-    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeOne());
-    assertThat(uc.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));
-    uc.addAllUnits(getUnitsOfDefaultPlayerOfUnitTypeTwo());
-    assertThat(uc.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(false)));
+    assertThat(unitCollection.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));
+    addAllUnitTypeOneUnitsOfDefaultPlayerToUnitCollection();
+    assertThat(unitCollection.allMatch(Matches.unitIsOfType(unitTypeOne)), is(equalTo(true)));

```

Comment:
This seems confusing why we have `addAllUnitTypeOneUnitsOfDefaultPlayerToUnitCollection();` if it does not change the value of the assertion. Is there an interaction between line 331 `addAllUnitTypeOneUnitsOfDefaultPlayerToUnitCollection();` and line 333 `addAllUnitTypeTwoUnitsOfDefaultPlayerToUnitCollection()`?

Could we delete line 331 and have the test still pass? Ultimately the problem is that there is magic happening on lines 331 and 333 with the calls to the 'void' return methods. To understand this function one has to understand that magic as well. As stated before, if what is happening is a data mutation, then returning the mutated state so a reader can see what has happened at a high level will make this easy to follow.